spdx-utils/src/main/kotlin/SpdxUtils.kt

```diff
@@ -23,7 +23,9 @@ import org.apache.commons.codec.binary.Hex
 import org.apache.commons.codec.digest.DigestUtils
 
 import java.io.File
+import java.io.FileNotFoundException
```

Comment:
Why is it tricky to get a list of LicenseRefs supported by scan-code?

Wouldn't it suffice to know the list of LicenseRefs scancode-toolkit has license refs for?
...Because that list would be a subset of the contents of this [1] directory?

I'm a bit afraid depending on such online resource because it can make ossreviewtoolkit results not reproducible and in the worst case break. Also does the content of that dir need to match with a specific revision of scan code? 

Considering above points I'd prefer to commit a copy of [1] to our repository and update it when needed, e.g. along with updating scan code itself. What do you think?

[1] https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/licenses

wondering if a hyrid solution would be worth it, but I'm not sure how static / dynamic the content of that dir is...

Reply:
> Wouldn't it suffice to know the list of LicenseRefs scancode-toolkit has license refs for?
> ...Because that list would be a subset of the contents of this [1] directory?

There are several not-so-nice (though absolutely technically solvable) implications due to the way ScanCode stores its licenses: The list of licenses ScanCode recognizes is implicitly defined by the list of license files in that directory, i.e. there is no single file whose contents we could simply read like we do for https://raw.githubusercontent.com/spdx/license-list-data/master/json/licenses.json. As iterating over remote repositories is not easily possible, we'd need to download all of the licenses. On top of that, there is no easy way to determine which of the licenses are core SPDX licenses, and which should be handled as LicenseRefs as ScanCode does not distinguish them by the file name, but only by the corresponding `.yml` file contents.

> I'm a bit afraid depending on such online resource because it can make ossreviewtoolkit results not reproducible and in the worst case break. Also does the content of that dir need to match with a specific revision of scan code?

We could easily avoid that by sticking to a tagged version of that directory.

> Considering above points I'd prefer to commit a copy of [1] to our repository and update it when needed, e.g. along with updating scan code itself. What do you think?

I'm against a copy, if by "copy" you really mean to manually copy & commit the files. That's too easy to mess with when updating, unless you automate the updating process, which adds complexity to the build system I'd like to avoid.

What I did consider is to add (the whole of) ScanCode as a submodule again, but that seems overkill to just get the license data, there were reasons to ditch that approach after all, and I like the clean policy we have right now that submodules are only required for running tests.

Reply:
By "making a copy" I meant some reliable / automated / not too error prone way to make a copy.
The copy tasks wouldn't need to execute with each and every build but just be manually run and the generated result commited. 

Did you consider making a copy using subversion, e.g. as mentioned here: 
https://stackoverflow.com/a/18194523  ?
..Maybe this is simple enough to not even integrate it into the build system to avoid the SVN dependency.

Reply:
Using Subversion at build time is still an involving but technically interesting approach. I'm looking at https://github.com/martoe/gradle-svntools-plugin/blob/master/docs/SvnExport.md now.